prov/tcp/src/tcpx_ep.c

```diff
@@ -483,11 +490,25 @@ static int tcpx_ep_getname(fid_t fid, void *addr, size_t *addrlen)
 	return (addrlen_in < *addrlen)? -FI_ETOOSMALL: FI_SUCCESS;
 }
 
+static int tcpx_ep_getpeer(struct fid_ep *ep, void *addr, size_t *addrlen)
+{
+	struct tcpx_ep *tcpx_ep;
+	size_t len;
+
+	tcpx_ep = container_of(ep, struct tcpx_ep, util_ep.ep_fid);
+	if (!tcpx_ep->dest_addr)
+		return -FI_EINVAL;
+	len = MIN(*addrlen, sizeof(struct sockaddr_in));
+	memcpy(addr, tcpx_ep->dest_addr, len);
+	*addrlen = sizeof(struct sockaddr_in);
+	return (len == sizeof(struct sockaddr_in)) ? 0 : -FI_ETOOSMALL;
```

Comment:
Why not just call getpeername() here and avoid the extra complexity of storing it?  We can pass the caller's addr/addrlen values directly to the call.

Also, IPv6 support is a requirement at this point.